src/sagas/task.saga.js

```diff
@@ -237,23 +249,40 @@ export function* taskRun({ task }: Action): Saga<void> {
 
       case 'exit':
         if (task.name === 'eject') {
+          // If ejecting failed due to unclean repo, then throw up a dialog
+          if (uncleanRepo) {
+            dialog.showMessageBox({
+              type: 'warning',
+              buttons: ['Ok'],
+              defaultId: 0,
+              cancelId: 0,
+              title: 'Oops!',
+              message: 'Dirty git state detected',
+              detail:
+                'Oh no! In order to eject, git state must be clean. Please commit your changes to continue ejecting.',
```

Comment:
I think we should change the last sentence from detail as the user needs to repeat eject after committing. Something like: `Oh no! In order to eject, git state must be clean. Please commit your changes and retry ejecting.`